spring-boot/src/main/java/com/sentrydemos/springboot/Application.java

```diff
@@ -44,24 +44,15 @@ class CustomTracesSamplerCallback implements TracesSamplerCallback {
 		public Double sample(SamplingContext context) {
 
 			CustomSamplingContext customSamplingContext = context.getCustomSamplingContext();
-			if (customSamplingContext != null) {
-				HttpServletRequest request = (HttpServletRequest) customSamplingContext.get("request");
-				return 1.0;
+
+		HttpServletRequest request = (HttpServletRequest) customSamplingContext.get("request");
```

Comment:
@bfpepper87 I just realized that line 48 will fail if the null check is not performed first. I see you moved the null check to down below to line 50, but I checked the current `master` code and it looks like the null check should be executed before your line 48 as well as before line 50. Could you re-arrange here? Or what do you think?

![image](https://user-images.githubusercontent.com/8920574/181724167-9cbaf942-023d-411a-a330-750323f2ef9f.png)

Reply:
the purpose of the null check is that so the following line won't fail to finish executing:
`HttpServletRequest request = (HttpServletRequest) customSamplingContext.get("request");`
With your most recent code, the null check occurs after that line, not before it (like in master). Unless I'm reading it wrong.